app/src/main/java/com/android/bookswap/data/source/network/MessageFirestoreSource.kt

```diff
@@ -229,64 +221,53 @@ class MessageFirestoreSource(private val db: FirebaseFirestore) : MessageReposit
    * @param context The context for displaying Toast messages.
    */
   override fun updateMessage(
-      message: DataMessage,
+      dataMessage: DataMessage,
       user1UUID: UUID,
       user2UUID: UUID,
       callback: (Result<Unit>) -> Unit
   ) {
     // Generate the chat path using the consistent `mergeUUIDs` order
     val chatPath = mergeUUIDs(user1UUID, user2UUID)
 
-    // Fetch the existing message to validate it exists and check the update window
     db.collection(COLLECTION_PATH)
         .document(chatPath)
         .collection("messages")
-        .document(message.uuid.toString())
+        .document(dataMessage.uuid.toString())
         .get()
         .addOnCompleteListener { task ->
-          if (task.isSuccessful && task.result != null && task.result!!.exists()) {
-            val document = task.result!!
-            val existingMessage = documentToMessage(document).getOrNull()
-
-            if (existingMessage != null) {
-              val currentTime = System.currentTimeMillis()
-
-              // Check if the message is within the 15-minute update window
-              if (currentTime - existingMessage.timestamp <= fifteenMinutesInMillis) {
-                // Prepare the updated fields
-                val updatedFields =
-                    mapOf(
-                        "text" to message.text,
-                        "timestamp" to currentTime,
-                        "messageType" to message.messageType.name)
-
-                // Perform the update
-                db.collection(COLLECTION_PATH)
-                    .document(chatPath)
-                    .collection("messages")
-                    .document(message.uuid.toString())
-                    .update(updatedFields)
-                    .addOnCompleteListener { updateTask ->
-                      if (updateTask.isSuccessful) {
-                        callback(Result.success(Unit))
-                      } else {
-                        callback(
-                            Result.failure(
-                                updateTask.exception ?: Exception("Error updating message")))
+          if (task.isSuccessful) {
+            val document = task.result
+            if (document != null && document.exists()) {
+              val existingMessage = documentToMessage(document).getOrNull()
+              if (existingMessage != null) {
+                val currentTime = System.currentTimeMillis()
+                if (currentTime - existingMessage.timestamp <= fifteenMinutesInMillis) {
+                  val messageMap = messageToDocument(dataMessage)
+                  db.collection(COLLECTION_PATH)
+                      .document(dataMessage.uuid.toString())
```

Comment:
Here you have reverted to the previous way of looking in the firebase for a message, it will not work, it is probably an error which arose when merging with main. Please have a look at it.